src/Workspaces/CoreTest/Execution/Extensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Execution
+{
+    internal static class Extensions
+    {
+        public static ChecksumObjects<ProjectChecksumObject> ToProjectObjects(this ChecksumCollection collection, ISolutionChecksumService service)
+        {
+            Contract.ThrowIfFalse(collection.Kind == WellKnownChecksumObjects.Projects);
+            return new ChecksumObjects<ProjectChecksumObject>(service, collection);
+        }
+
+        public static ChecksumObjects<DocumentChecksumObject> ToDocumentObjects(this ChecksumCollection collection, ISolutionChecksumService service)
+        {
+            Contract.ThrowIfFalse(collection.Kind == WellKnownChecksumObjects.Documents || collection.Kind == WellKnownChecksumObjects.TextDocuments);
+            return new ChecksumObjects<DocumentChecksumObject>(service, collection);
+        }
+    }
+
+    internal class ChecksumObjects<T> : ChecksumObject where T : ChecksumObject
```

Comment:
we mean this - ChecksumObjects<T> ?

this is just helper type I created in unit test to retrieve item from checksumCollection. it is not production code.

I didn't think much about name, I just put something there. if confusing, I can name it PackChecksumAndObjects something like that.